### ISSUE TITLE
No warning/error on unclosed option tag

### ISSUE THREAD

**Issue by dagnelies** [ractivejs/ractive#2466]
See http://jsfiddle.net/xc2o8y08/61/

**Comment by fskreuz** [comment]
Browsers are lenient. When it sees invalid HTML like this, it tries to make sense of it. In this case, it would auto-close `<option>`.

However, to save us hair, we could prevent this in the parser. Here's a list of [void elements](https://www.w3.org/TR/html5/syntax.html#void-elements). `<option>` isn't one.

**Comment by evs-chris** [comment]
Yeah, like with `<li>`, I think this is one of those things we can't really prevent entirely because someone will trip over the other side of it at some point. We generally stick with the spec when the question arises, and the spec now says they're not required.

I still struggle with that because I cut my teeth on good old xhtml and think that not closing tags is ugly and probably a mortal sin. Speaking of mortal sins, void elements _must_ have a self-closing `/`, I tell you!

I'd not be opposed to adding a flag for warnings in parser, though.

**Comment by Rich-Harris** [comment]
Yeah, the parser sides with browsers on this one. The only time it will *always* warn you about an unclosed tag is if an element is left open and it reaches the end of the template.

Perhaps there should be a `strict` mode for the parser that doesn't let you implicitly close elements?

**Comment by dagnelies** [comment]
that would be nice :+1:

**Comment by dagnelies** [comment]
as another detail, it would be nice if `strict` would log a warning when a computed variable throws an exception.
